spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/CleanPathFilter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.filter;
+
+import java.net.URI;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.core.Ordered;
+import org.springframework.util.StringUtils;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * @author Ryan Baxter
+ */
+public class CleanPathFilter implements GlobalFilter, Ordered {
```

Comment:
What is the purpose of the clean path filter? Can we have a small comment giving an example of what it does?

Reply:
The purpose is to make sure that the path added to the URI we end up adding to the URI of the route is "clean".  I can certainly add some comments!